cli/internal/sync/map.go

```diff
@@ -0,0 +1,63 @@
+package sync
+
+import (
+	"sync"
+)
+
+// Map is a wrapper around sync.Map that provides type safety for generics.
+// Map is like a Go map[any]any but is safe for concurrent use
+// by multiple goroutines without additional locking or coordination.
+// Loads, stores, and deletes run in amortized constant time.
+type Map[K, V any] struct {
+	Untyped sync.Map
+}
+
+// Delete deletes the value for a key.
+func (m *Map[K, V]) Delete(key K) { m.Untyped.Delete(key) }
+
+// Load returns the value stored in the map for a key, or nil if no
+// value is present.
+// The ok result indicates whether value was found in the map.
+func (m *Map[K, V]) Load(key K) (value V, ok bool) {
+	v, ok := m.Untyped.Load(key)
+	if !ok {
+		return value, ok
+	}
+	return v.(V), ok
```

Comment:
I think you need type assertions on these returns, right?

Reply:
I made the map unexported now so it is impossible to write into it in a case where the return type can not be cast.